src/algebra/category/FinVect.lean

```diff
@@ -0,0 +1,194 @@
+/-
+Copyright (c) 2021 Jakob von Raumer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob von Raumer
+-/
+import linear_algebra.finite_dimensional
+import category_theory.monoidal.rigid
+import linear_algebra.finsupp_vector_space
+import linear_algebra.dual
+import linear_algebra.coevaluation
+import .Module.monoidal
+
+/-!
+# The category of finite dimensional vector spaces
+
+This introduces `FinVect K`, the category of finite dimensional vector spaces on a field `K`.
+It is implemented as a full subcategory on a subtype of  `Module K`.
+We first create the instance as a category, then as a monoidal category and then as a rigid monoidal
+category.
+
+## Future work
+
+* Show that `FinVect K` is a symmetric monoidal category.
+
+-/
+noncomputable theory
+
+open category_theory Module.monoidal_category
+open_locale classical big_operators
+
+universes u
+
+variables (K : Type u) [field K]
+
+/-- Define `FinVect` as the subtype of `Module.{u} K` of finite dimensional vector spaces. -/
+def FinVect := { V : Module.{u} K // finite_dimensional K V.carrier }
+
+namespace FinVect
+
+instance category : category (FinVect K) := by unfold FinVect; apply_instance
+
+instance finite_dimensional (V : FinVect K): finite_dimensional K V.val := V.prop
```

Comment:
Then remove the `.val` here, and ... many more below! Can I leave it to you to experiment?